### ISSUE TITLE
Avoid admin UI crashes when data of block does not contain a type

### ISSUE THREAD

**Issue by alexander-schranz** [sulu/sulu#6197]
| Q | A
| --- | ---
| Sulu Version | 2.3-dev
| PHP Version | 7.4.12
| DB Version | MySQL 5.7.35
| Browser Version | 

# Actual Behavior

Currently if you are having a custom form which uses blocks and the API does not provide a `type` property inside it the admin UI crashes with:

> It is impossible that a block has no type. This should not happen and is likely a bug.

https://github.com/sulu/sulu/blob/20dc236ee567d79c4940107309a03569ba84a51b/src/Sulu/Bundle/AdminBundle/Resources/js/containers/FieldBlocks/FieldBlocks.js#L237-L241

# Expected Behavior

The admin JS should not crash when no `type` is provided. There are usecases where the type is not necessary as example the block is used on a entity for a to many relation and only 1 type exist.

Currently if the if a false block type is provided a fallback to the default type is done: https://github.com/sulu/sulu/blob/20dc236ee567d79c4940107309a03569ba84a51b/src/Sulu/Bundle/AdminBundle/Resources/js/containers/FieldBlocks/FieldBlocks.js#L263. This should be I think the same when the api does not return any type.

# Possible Solutions

Always fallback to the default type if no type is given

**Comment by mamazu** [comment]
Yeah, especially when changing the XML configuration without keeping the data in mind this makes it hard.

I'd like to try to tackle this.

**Comment by mamazu** [comment]
When trying to reproduce it, (and remove the type of the currently used block). It always removes the block entirely. Is that intentional behaviour?

**Comment by draconivis** [comment]
I got the same result as @mamazu when reproducing this. The block just doesn't show up after deleting the type (but is visible again after adding it back).
